File: output/xCodeEval_remove_8000/C++/Java/5ff06312c9f00109c984ee2127a9080f.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class 5ff06312c9f00109c984ee2127a9080fxcodeeval_processed_8000.json {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] input = br.readLine().split(" ");
        int l = Integer.parseInt(input[0]);
        int r = Integer.parseInt(input[1]);

        System.out.println(ans(r) - ans(l - 1));
    }

    public static boolean prime(int x) {
        if (x < 2)
            return false;
        int j = 2;
        while (j * j <= x) {
            if (x % j == 0) {
                return false;
            }
            if (j == 2)
                j++;
            else
                j += 2;
        }
        return true;
    }

    public static int ans(int x) {
        int j = x / 100000;
        int ret = 0;
        if (x >= 2)
            ret++;
        for (int i = 0; i <= j; ++i) {
            ret += f[i];
        }
        if (j > 0)
            ret--;

        int b = j * 100000;
        b++;

        for (int i = b; i <= x; i += 4) {
            if (prime(i))
                ret++;
        }
        return ret;
    }

    static final int[] f = {

File: output/GPT3.5-Turbo/xCodeEval_remove_4000/C++/Python/5ff06312c9f00109c984ee2127a9080f.py
import math

def prime(x):
    if x < 2:
        return False
    j = 2
    while j * j <= x:
        if x % j == 0:
            return False
        if j == 2:
            j += 1
        else:
            j += 2
    return True

def ans(x):
    j = x // 100000
    ret = 0
    if x >= 2:
        ret += 1
    for i in range(j + 1):
        ret += f[i]
    if j > 0:
        ret -= 1

    b = j * 100000
    b += 1
    
    for i in range(b, x + 1, 4):
        if prime(i):
            ret += 1
    return ret

l, r = map(int, input().split())

print(ans(r) - ans(l - 1))
